### DISCUSSION TITLE
Different labeling on different systems

### DISCUSSION THREAD

**Discussion by 4tler** [mapsforge/mapsforge#1219] [Q&A] [answered]
Hello, I have a problem with the labeling in area elements. 

When I view the theme via the Cruiser application on the PC, the labeling of the areas is as I want. However, on mobile the labeling is constantly repeated. The more I zoom in the more often the label appears.

my theme.xml:
```  
 <!-- Campingplatz -->
    <rule e="way" k="tourism" v="camp_site" closed="yes" zoom-min="10">
        <rule e="any" k="*" v="*">
            <area fill="#DFE6B8" />
        </rule>
        <rule cat="accommodation" e="any" k="*" v="*" zoom-min="9" zoom-max="15">
            <symbol id="camping" display="ifspace" src="file:/symbols/tourism_campsite.svg" symbol-width="20" />
        </rule>
        <rule cat="accommodation" e="any" k="*" v="*" zoom-min="14" zoom-max="19">
            <caption k="name" symbol-id="camping" display="ifspace" position="auto" font-style="bold" font-size="10" fill="#FF4D00" stroke="#E5FFFFFF" stroke-width="2" />
        </rule>
    </rule> 

```

with mobile:
![image](https://user-images.githubusercontent.com/74738662/120937219-9fd4f200-c70c-11eb-9aa0-96144104fe72.png)
with CRUISER:
![image](https://user-images.githubusercontent.com/74738662/120937243-bda25700-c70c-11eb-9b82-e5871cfc08d2.png)

**Comment by devemux86** [comment]
Do you use [label layer](https://github.com/mapsforge/mapsforge/blob/master/docs/LabelLayer.md)?
What settings do you use in Cruiser?

It is better to make such comparisons with repository samples, so that everyone can test the report.
- [mapsforge-samples-android](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-android)
- [mapsforge-samples-awt](https://github.com/mapsforge/mapsforge/tree/master/mapsforge-samples-awt)

**Comment by 4tler** [comment]
I use Cruiser with the default settings and a self-generated .map file + custom theme. 

I downloaded the Mapsforge example app and included my map + theme. The display was flawless as I had hoped.

I did the same with the mapsforge/vtm example app and the display was again bad. 

In our app we also use the vtm library with a label-layer.

Are there any differences in the rendering of mapsforge data with the vtm library? Is the only solution now to switch to mapsforge because vtm does not fully support it? 

Thanks for the support.

example vtm:
![image](https://user-images.githubusercontent.com/74738662/121079747-d5481100-c7da-11eb-9546-b86309011e99.png)

example mapsforge:
![image](https://user-images.githubusercontent.com/74738662/121080025-2eb04000-c7db-11eb-9de3-19f096ecafbc.png)

**Comment by devemux86** [comment, marked as answer]
Which library do you use, [Mapsforge](https://github.com/mapsforge/mapsforge) or [VTM](https://github.com/mapsforge/vtm)?
If you need something for VTM, please create a discussion in its repository.

The libraries are different, you can use the one that best suits your needs.

---

- Cruiser uses Mapsforge
- Cruiser GL uses VTM

**Reply by 4tler** [reply]
We use the vtm library. It was probably a mistake to think that vtm includes an integration of mapsforge and both work the same. We assumed that vtm supports other formats like mapbox vector tiles etc. in contrast to the pure mapsforge library. But it seems to be a different way of processing (displaying) the map files.

I will open a new discussion in vtm to get tips for labeling. 

PS: I am correct in assuming that a theme written for mapsforge will basically work in vtm as well, right?

**Comment by devemux86** [comment]
> I am correct in assuming that a theme written for mapsforge will basically work in vtm as well

It can read it, but you better use VTM themes in VTM and Mapsforge themes in Mapsforge.
VTM themes are more advanced and contain additional rules.
